src/main/java/dan200/computercraft/api/network/Packet.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * This file is part of ComputerCraft - http://www.computercraft.info
+ * Copyright Daniel Ratcliffe, 2011-2017. Do not distribute without permission.
+ * Send enquiries to dratcliffe@gmail.com
+ */
+package dan200.computercraft.api.network;
+
+import com.google.common.base.Preconditions;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * Represents a packet which may be sent across a {@link IPacketNetwork}.
+ *
+ * @see IPacketSender
+ * @see IPacketNetwork#transmit(Packet)
+ * @see IPacketReceiver#receiveDifferentDimension(Packet)
+ * @see IPacketReceiver#receiveSameDimension(Packet, double)
+ */
+public class Packet
+{
+    private final int m_channel;
+    private final int m_replyChannel;
+    private final Object m_payload;
+
+    private final double m_range;
```

Comment:
Should "range" and "interdimensional" be properties on the packet itself, or just additional arguments to transmit?